### PR TITLE
Add lut-based functions and benches

### LINKED ISSUE #1: Explore using LUT's instead of arithmetic in parsing and displaying
Explore if lut based counterparts to current parsing and displaying functions would be faster than what we're currently doing. If so, use them instead.

### PR DESCRIPTION
**Pull request by fstasiak**
resolves #1

### CODE REVIEW

**Comment by fstasiak** [comment]
# Benches

Benchmarks were performed on Ryzen 5800X.

n: length of `HexString` - `HexString<n>`

Parsing using `try_parse` (both lowercase and uppercase supported) vs `try_parse_lut`, input string contains only lowercase characters:
| n     |    old    |    new    | change |
|-------|:---------:|:---------:|:------:|
| 16    | 47.414 ns | 26.613 ns |  -44%  |
| 64    | 324.59 ns | 93.480 ns |  -71%  |
| 256   | 1316.0 ns | 340.19 ns |  -74%  |
| 1024  | 4886.1 ns | 1423.5 ns |  -71%  |
| 4096  |  19019 ns | 5401.7 ns |  -72%  |
| 16384 |  82234 ns |  23359 ns |  -72%  |

Parsing using `try_parse` (both lowercase and uppercase supported) vs `try_parse_lut`, input string contains only uppercase characters:
| n     |    old    |    new    | change |
|-------|:---------:|:---------:|:------:|
| 16    | 50.453 ns | 26.464 ns |  -48%  |
| 64    | 331.83 ns | 94.931 ns |  -71%  |
| 256   | 1373.0 ns | 342.04 ns |  -75%  |
| 1024  | 4849.3 ns | 1437.8 ns |  -70%  |
| 4096  |  19342 ns | 5389.8 ns |  -72%  |
| 16384 |  84991 ns |  23539 ns |  -72%  |

Parsing using `try_parse_lower` vs `try_parse_lower_lut`:
| n     |    old    |    new    | change |
|-------|:---------:|:---------:|:------:|
| 16    | 28.545 ns | 26.435 ns |   -7%  |
| 64    | 364.57 ns | 93.634 ns |  -74%  |
| 256   | 1599.2 ns |  414.8 ns |  -74%  |
| 1024  | 6390.9 ns | 1427.0 ns |  -78%  |
| 4096  |  24348 ns | 5784.6 ns |  -76%  |
| 16384 |  92377 ns |  23390 ns |  -75%  |

Parsing using `try_parse_upper` vs `try_parse_upper_lut`:
| n     |    old    |    new    | change |
|-------|:---------:|:---------:|:------:|
| 16    | 28.942 ns | 25.961 ns |  -10%  |
| 64    | 384.78 ns | 104.56 ns |  -73%  |
| 256   | 1497.5 ns | 362.81 ns |  -76%  |
| 1024  | 6390.9 ns | 1427.0 ns |  -78%  |
| 4096  |  24316 ns | 6639.6 ns |  -73%  |
| 16384 |  87324 ns |  26908 ns |  -69%  |

Displaying using `to_upper` vs `to_upper_lut`:
| n     |    old    |    new    | change |
|-------|:---------:|:---------:|:------:|
| 16    | 114.16 ns | 114.44 ns |   0%   |
| 64    | 386.88 ns | 389.33 ns |   +1%  |
| 256   | 1478.7 ns | 1493.3 ns |   +1%  |
| 1024  | 5869.7 ns | 5929.5 ns |   +1%  |
| 4096  |  23397 ns | 233591 ns |   +1%  |
| 16384 |  93675 ns |  94228 ns |   +1%  |

Benchmark result of `to_lower` vs `to_lower_lut` has been omitted as its results are identical to those of `to_upper`.

---
To sum up, big gains when parsing, no improvement when displaying.
